components/timestamp/index.ts

```diff
@@ -21,29 +21,33 @@ type Props = {
     userTimezone?: UserTimezone;
 }
 
-function mapStateToProps(state: GlobalState, {userTimezone}: Props) {
-    const currentUserId = getCurrentUserId(state);
+function makeMapStateToProps() {
+    const getUserTimezone = makeGetUserTimezone();
 
-    let timeZone: TimestampProps['timeZone'];
-    let hourCycle: TimestampProps['hourCycle'];
-    let hour12: TimestampProps['hour12'];
+    return (state: GlobalState, {userTimezone}: Props) => {
+        const currentUserId = getCurrentUserId(state);
 
-    if (areTimezonesEnabledAndSupported(state)) {
-        timeZone = getUserCurrentTimezone(userTimezone ?? getUserTimezone(state, currentUserId)) ?? undefined;
-    }
+        let timeZone: TimestampProps['timeZone'];
+        let hourCycle: TimestampProps['hourCycle'];
+        let hour12: TimestampProps['hour12'];
 
-    const useMilitaryTime = getBool(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.USE_MILITARY_TIME, false);
+        if (areTimezonesEnabledAndSupported(state)) {
+            timeZone = getUserCurrentTimezone(userTimezone ?? getUserTimezone(state, currentUserId)) ?? undefined;
```

Comment:
What's the relationship between `getUserCurrentTimezone` and `getUserTimezone` now? Maybe that's something to solve in another PR, though since it doesn't seem to be a performance concern

Reply:
Oh, `getUserTimezone` returns the object with the additional information about the user's timezone settings while `getUserCurrentTimezone` breaks that down into their actual timezone which can be used for formatting Dates